gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/client/impl/ClientManagerImpl.java

```diff
@@ -79,12 +89,16 @@ public void afterPropertiesSet() throws Exception {
     public void onEvent(Event<ApplicationEvent, Payload> event) {
         if (event.content().getReferenceType() == ReferenceType.DOMAIN &&
                 (domain.isMaster() || domain.getId().equals(event.content().getReferenceId()))) {
+            // count the event after the test to avoid duplicate events across domains
+            appEvtCounter.increment();
             switch (event.type()) {
                 case DEPLOY:
+                    appGauge.incrementValue();
                 case UPDATE:
                     deployClient(event.content().getId());
                     break;
                 case UNDEPLOY:
```

Comment:
Shouldn't we decrement after the remove is done ?

Reply:
FP